vault/tests/test_vault.py

```diff
@@ -104,6 +105,21 @@ def test_service_check_500_fail(self, aggregator, global_tags):
 
         aggregator.assert_service_check(Vault.SERVICE_CHECK_CONNECT, status=Vault.CRITICAL, tags=global_tags, count=1)
 
+    def test_api_unreachable(self, aggregator, global_tags):
+        instance = INSTANCES['main']
+        c = Vault(Vault.CHECK_NAME, {}, [instance])
+
+        with mock.patch('requests.get', return_value=MockResponse('', exception=requests.exceptions.RequestException)):
```

Comment:
You can do this instead
```suggestion
        with mock.patch('requests.get', side_effect=requests.exceptions.RequestException)):
```
and no need to update the MockResponse class then